### PR TITLE
usbhost feature support for stm32F4 discovery board

### PR DESCRIPTION
**Pull request by satheesh-ks**
Issue: usb pen-drive not detected in stm32f407-disc1 board

Cause: latest usbhost changes are missing from "drivers/usbhost/usbhost_enumerate.c".
Synced the enumeration and stm32 chip code to match with latest nuttX code.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ea50ac590b6e2bb2cf8ea964502831858a8e1930] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ea50ac590b6e2bb2cf8ea964502831858a8e1930] - Code Rule Check Result:
os/arch/arm/src/stm32/chip/stm32_otgfs.h:486: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/stm32/stm32_otgfsdev.c:170: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:170: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:173: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:173: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:174: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:174: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:175: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:175: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:176: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:176: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:177: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:177: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:178: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:178: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:179: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:179: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:180: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:180: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:181: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:181: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:182: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:182: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:183: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:183: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:184: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:184: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:185: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:185: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:186: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:186: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:187: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:187: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:191: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:191: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:192: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:192: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:195: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:195: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:196: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:196: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:197: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:197: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:198: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:198: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:199: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:199: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:200: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:200: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:201: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:201: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:202: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:202: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:203: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:203: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:204: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:204: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:205: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:205: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:206: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:206: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:207: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfsdev.c:207: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/stm32/stm32_otgfsdev.c:457: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:458: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:459: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:460: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:461: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:462: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:481: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:482: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:483: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:484: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:485: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:486: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:488: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:489: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_otgfsdev.c:490: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:335: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:336: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:337: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:338: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:358: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:359: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/board/stm32f407-disc1/src/stm32_usb.c:185: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by sunghan-chang** [comment]
@satheesh-ks Please find coding rule violations first.

**Comment by satheesh-ks** [comment]
> @satheesh-ks Please find coding rule violations first.

@sunghan-chang I will try fix it. If I have the rule-checker-tool I verify it locally before making a PR. where I can get it?

I used "./tools/cformatter.sh" before pushing the changes.

**Comment by sunghan-chang** [comment]
@satheesh-ks SE team merged it into SVACE tool. It's only way to run it.

**Comment by sunghan-chang** [comment]
@satheesh-ks Why do you write ` for stm32 discovery board ` in second commit title?
Is it stm specific commit?

**Comment by satheesh-ks** [comment]
> @satheesh-ks Why do you write `for stm32 discovery board` in second commit title?
> Is it stm specific commit?

@sunghan-chang Corrected the commit title as "os/drivers/usbhost: Fix usbhost enumerate failure"

**Comment by seinfra** [comment]
Target : [7d9e5aa692ef1a3946f8f6849c8f38843535547c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7d9e5aa692ef1a3946f8f6849c8f38843535547c] - Code Rule Check Result:
os/arch/arm/src/stm32/stm32_usbdev.c:358: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/stm32/stm32_usbdev.c:359: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [29abf4f58097371393fa0ce39564d4398491454b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [29abf4f58097371393fa0ce39564d4398491454b] - Code Rule Check OK.

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 1]
```diff
@@ -1012,10 +1055,10 @@ static void stm32_txfifo_write(FAR struct stm32_ep_s *privep, FAR uint8_t *buf,
...
 		 */
 
-		regval = (uint32_t)*buf++;
-		regval |= ((uint32_t)*buf++) << 8;
-		regval |= ((uint32_t)*buf++) << 16;
-		regval |= ((uint32_t)*buf++) << 24;
+		regval = (uint32_t) * buf++;
+		regval |= ((uint32_t) * buf++) << 8;
+		regval |= ((uint32_t) * buf++) << 16;
+		regval |= ((uint32_t) * buf++) << 24;
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: please revert these changes. This is abnormal change of formatter. Previous codes are right.

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 2]
```diff
@@ -1058,13 +1101,13 @@ static void stm32_epin_transfer(FAR struct stm32_ep_s *privep, FAR uint8_t *buf,
 		 *   perform the transfer.
 		 */
 
-		pktcnt = ((uint32_t)nbytes + (privep->ep.maxpacket - 1)) / privep->ep.maxpacket;
+		pktcnt = ((uint32_t) nbytes + (privep->ep.maxpacket - 1)) / privep->ep.maxpacket;
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: same as previous

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 3]
```diff
@@ -1058,13 +1101,13 @@ static void stm32_epin_transfer(FAR struct stm32_ep_s *privep, FAR uint8_t *buf,
...
 	}
 
 	/* Set the XFRSIZ and PKTCNT */
 
 	regval |= (pktcnt << OTGFS_DIEPTSIZ_PKTCNT_SHIFT);
-	regval |= ((uint32_t)nbytes << OTGFS_DIEPTSIZ_XFRSIZ_SHIFT);
+	regval |= ((uint32_t) nbytes << OTGFS_DIEPTSIZ_XFRSIZ_SHIFT);
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: same as previous

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 4]
```diff
@@ -1266,7 +1309,7 @@ static void stm32_epin_request(FAR struct stm32_usbdev_s *priv, FAR struct stm32
 
 		regval = stm32_getreg(regaddr);
 		if ((int)(regval & OTGFS_DTXFSTS_MASK) < nwords) {
-			usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_EMPWAIT), (uint16_t)regval);
+			usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_EMPWAIT), (uint16_t) regval);
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: same as previous

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 5]
```diff
@@ -2134,7 +2177,7 @@ static inline void stm32_ep0out_stdrequest(struct stm32_usbdev_s *priv, FAR stru
 			 * the completion of the status phase.
 			 */
 
-			stm32_setaddress(priv, (uint16_t)priv->ctrlreq.value[0]);
+			stm32_setaddress(priv, (uint16_t) priv->ctrlreq.value[0]);
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: same as previous

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 6]
```diff
@@ -2206,7 +2249,7 @@ static inline void stm32_ep0out_stdrequest(struct stm32_usbdev_s *priv, FAR stru
 			 */
 
 			if (ret == OK) {
-				uint8_t cfg = (uint8_t)ctrlreq->value;
+				uint8_t cfg = (uint8_t) ctrlreq->value;
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: same as previous

File: os/arch/arm/src/stm32/stm32_otgfsdev.c [hunk 7]
```diff
@@ -2740,14 +2783,14 @@ static inline void stm32_epin_interrupt(FAR struct stm32_usbdev_s *priv)
...
-				usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_EPDISD), (uint16_t)diepint);
+				usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_EPDISD), (uint16_t) diepint);
 				stm32_putreg(OTGFS_DIEPINT_EPDISD, STM32_OTGFS_DIEPINT(epno));
 			}
 #endif
 			/* Transmit FIFO empty */
 
 			if ((diepint & OTGFS_DIEPINT_TXFE) != 0) {
-				usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_TXFE), (uint16_t)diepint);
+				usbtrace(TRACE_INTDECODE(STM32_TRACEINTID_EPIN_TXFE), (uint16_t) diepint);
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfsdev.c]: Could you check all of this kind of change?

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 1 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 2 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 3 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 4 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 5 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 6 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: done

File: os/arch/arm/src/stm32/stm32_otgfsdev.c (hunk 7 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfsdev.c]: All such changes has been taken care now

**Comment by seinfra** [comment]
Target : [f17acafcef3fd677814297b2f1201fc0d94b6ff0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f17acafcef3fd677814297b2f1201fc0d94b6ff0] - Code Rule Check OK.

File: os/arch/arm/src/stm32/stm32_otgfshost.c [hunk 8]
```diff
@@ -950,7 +1026,7 @@ static int stm32_chan_wait(FAR struct stm32_usbhost_s *priv, FAR struct stm32_ch
 		 * awakened by a signal too.
 		 */
 
-		DEBUGASSERT(ret == OK || get_errno() == EINTR);
+		DEBUGASSERT(ret == OK || ret == -EINTR);
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32/stm32_otgfshost.c]: sem_wait does not return errno. It returns OK or ERROR . Previous code is right.

File: os/arch/arm/src/stm32/stm32_otgfshost.c [hunk 9]
```diff
@@ -768,23 +802,23 @@ static void stm32_chan_configure(FAR struct stm32_usbhost_s *priv, int chidx)
 
 	/* Program the HCCHAR register */
 
-	regval = ((uint32_t)priv->chan[chidx].maxpacket << OTGFS_HCCHAR_MPSIZ_SHIFT) | ((uint32_t)priv->chan[chidx].epno << OTGFS_HCCHAR_EPNUM_SHIFT) | ((uint32_t)priv->chan[chidx].eptype << OTGFS_HCCHAR_EPTYP_SHIFT) | ((uint32_t)priv->devaddr << OTGFS_HCCHAR_DAD_SHIFT);
+	regval = ((uint32_t) chan->maxpacket << OTGFS_HCCHAR_MPSIZ_SHIFT) | ((uint32_t) chan->epno << OTGFS_HCCHAR_EPNUM_SHIFT) | ((uint32_t) chan->eptype << OTGFS_HCCHAR_EPTYP_SHIFT) | ((uint32_t) chan->funcaddr << OTGFS_HCCHAR_DAD_SHIFT);
```
Comment (yashwanth686007) [inline os/arch/arm/src/stm32/stm32_otgfshost.c]: Remove the space between typecasting and variable.

File: os/arch/arm/src/stm32/stm32_otgfshost.c [hunk 10]
```diff
@@ -673,6 +706,7 @@ static inline void stm32_chan_freeall(FAR struct stm32_usbhost_s *priv)
 
 static void stm32_chan_configure(FAR struct stm32_usbhost_s *priv, int chidx)
 {
+	FAR struct stm32_chan_s *chan = &priv->chan[chidx];
```
Comment (yashwanth686007) [inline os/arch/arm/src/stm32/stm32_otgfshost.c]: No need to add checks for input variables?

File: os/arch/arm/src/stm32/stm32_otgfshost.c [hunk 11]
```diff
@@ -2196,7 +2816,7 @@ static inline void stm32_gint_rxflvlisr(FAR struct stm32_usbhost_s *priv)
 		if (bcnt > 0 && priv->chan[chidx].buffer != NULL) {
 			/* Transfer the packet from the Rx FIFO into the user buffer */
 
-			dest = (FAR uint32_t *)priv->chan[chidx].buffer;
+			dest = (FAR uint32_t *) priv->chan[chidx].buffer;
```
Comment (yashwanth686007) [inline os/arch/arm/src/stm32/stm32_otgfshost.c]: Same as above.

File: os/arch/arm/src/stm32/stm32_otgfshost.c (hunk 8 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfshost.c]: Done

File: os/arch/arm/src/stm32/stm32_otgfshost.c (hunk 9 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfshost.c]: Done

File: os/arch/arm/src/stm32/stm32_otgfshost.c (hunk 10 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfshost.c]: Done

File: os/arch/arm/src/stm32/stm32_otgfshost.c (hunk 11 above)
Comment (satheesh-ks) [reply os/arch/arm/src/stm32/stm32_otgfshost.c]: Done

**Comment by seinfra** [comment]
Target : [4419f8bfcbab6354a9a12b0ade8d5ea911ca4696] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4419f8bfcbab6354a9a12b0ade8d5ea911ca4696] - Code Rule Check Result:
os/arch/arm/src/stm32/stm32_otgfshost.c:712: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/stm32/stm32_otgfshost.c:714: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/stm32/stm32_otgfshost.c:714: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: os/include/tinyara/usb/usb.h [hunk 12]
```diff
@@ -264,6 +264,22 @@
 #define USBOTG_FEATURE_A_HNP_SUPPORT            4	/* A device supports HNP */
 #define USBOTG_FEATURE_A_ALT_HNP_SUPPORT        5	/* Another port on the A device supports HNP */
 
+#define USB_SPEED_UNKNOWN                       0	/* Transfer rate not yet set */
+#define USB_SPEED_LOW                           1	/* USB 1.1 */
```
Comment (NilabhShant) [inline os/include/tinyara/usb/usb.h]: USB low speed version was 1.0 which has transfer rate of 1.5 Mbits/s
and then comes USB full speed 1.1 has transfer rate of 12Mbit/s

please do 1.1 --> 1.0

**Comment by seinfra** [comment]
Target : [289a0c0270d7bb2bee423dd7f2b650eb048ac5df] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [289a0c0270d7bb2bee423dd7f2b650eb048ac5df] - Code Rule Check OK.

File: os/include/tinyara/usb/usbhost.h [hunk 13]
```diff
@@ -108,23 +121,21 @@
  *
  ************************************************************************************/
 
-#define CLASS_CREATE(reg, drvr, id) ((reg)->create(drvr, id))
+#define CLASS_CREATE(reg,hport,id) ((reg)->create(hport,id))
```
Comment (NilabhShant) [inline os/include/tinyara/usb/usbhost.h]: Coding rules: there should be a space after comma .

File: os/include/tinyara/usb/usbhost.h [hunk 14]
```diff
@@ -166,36 +176,34 @@
...
+ *   nature of the failure
  *
  * Assumptions:
  *   - Called from a single thread so no mutual exclusion is required.
  *   - Never called from an interrupt handler.
  *
  *******************************************************************************/
 
-#define CONN_WAIT(conn, connected) ((conn)->wait(conn, connected))
+#define CONN_WAIT(conn,hport) ((conn)->wait(conn,hport))
```
Comment (NilabhShant) [inline os/include/tinyara/usb/usbhost.h]: same as above.

File: os/include/tinyara/usb/usbhost.h [hunk 15]
```diff
@@ -251,7 +262,7 @@
  *
  ************************************************************************************/
 
-#define DRVR_EP0CONFIGURE(drvr, funcaddr, mps) ((drvr)->ep0configure(drvr, funcaddr, mps))
+#define DRVR_EP0CONFIGURE(drvr,ep0,funcaddr,speed,mps) ((drvr)->ep0configure(drvr,ep0,funcaddr,speed,mps))
```
Comment (NilabhShant) [inline os/include/tinyara/usb/usbhost.h]: same as above, Please check for space after comma at all places in this file.

File: os/include/tinyara/usb/usbhost.h [hunk 16]
```diff
@@ -587,21 +681,74 @@ struct usbhost_registry_s {
...
+
+	FAR struct usbhost_hubport_s *hport;	/* The port used by this class instance */
+
 	/* Provides the configuration descriptor to the class.  The configuration
 	 * descriptor contains critical information needed by the class in order to
 	 * initialize properly (such as endpoint selections).
 	 */
 
-	int (*connect)(FAR struct usbhost_class_s *usbclass, FAR const uint8_t *configdesc, int desclen, uint8_t funcaddr);
+	int (*connect)(FAR struct usbhost_class_s *devclass, FAR const uint8_t *configdesc, int desclen /*, uint8_t funcaddr */);
```
Comment (NilabhShant) [inline os/include/tinyara/usb/usbhost.h]: If funcaddr parameter is not needed can't we remove it fully.

**Comment by sunghan-chang** [comment]
@satheesh-ks When I see SRIB another's approval, I will merge this. Could you check it with your co-worker.

File: os/include/tinyara/usb/usb.h (hunk 12 above)
Comment (satheesh-ks) [reply os/include/tinyara/usb/usb.h]: Done

File: os/include/tinyara/usb/usbhost.h (hunk 13 above)
Comment (satheesh-ks) [reply os/include/tinyara/usb/usbhost.h]: Done for all the occurrences

File: os/include/tinyara/usb/usbhost.h (hunk 14 above)
Comment (satheesh-ks) [reply os/include/tinyara/usb/usbhost.h]: Done for all the occurrences

File: os/include/tinyara/usb/usbhost.h (hunk 15 above)
Comment (satheesh-ks) [reply os/include/tinyara/usb/usbhost.h]: Done for all the occurrences

File: os/include/tinyara/usb/usbhost.h (hunk 16 above)
Comment (satheesh-ks) [reply os/include/tinyara/usb/usbhost.h]: Removed

**Comment by seinfra** [comment]
Target : [ba0f5880bf5757e1b6daa322b2533624a21e5113] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ba0f5880bf5757e1b6daa322b2533624a21e5113] - Code Rule Check OK.
